cli/src/cmd/deploy.js

```diff
@@ -10,30 +10,47 @@ import { readPackageJonConfig } from '../utils/iAppConfigFile.js';
 import { askForDockerhubAccessToken } from '../cli-helpers/askForDockerhubAccessToken.js';
 import { handleCliError } from '../cli-helpers/handleCliError.js';
 import { getSpinner } from '../cli-helpers/spinner.js';
+import { askForAppSecret } from '../cli-helpers/askForAppSecret.js';
+import { askForWalletPrivateKey } from '../cli-helpers/askForWalletPrivateKey.js';
+import { Wallet } from 'ethers';
+import { getIExecDebug } from '../utils/iexec.js';
 
 export async function deploy() {
   const spinner = getSpinner();
+  try {
+    const dockerhubUsername = await askForDockerhubUsername({ spinner });
+    const dockerhubAccessToken = await askForDockerhubAccessToken({ spinner });
+
+    const { iAppVersion } = await spinner.prompt([
+      {
+        type: 'input',
+        name: 'iAppVersion',
+        message: 'What is the version of your iApp?',
+        default: '0.1.0',
+      },
+    ]);
 
-  const dockerhubUsername = await askForDockerhubUsername({ spinner });
-  const dockerhubAccessToken = await askForDockerhubAccessToken({ spinner });
+    const appSecret = await askForAppSecret({ spinner });
 
-  const { iAppVersion } = await spinner.prompt([
-    {
-      type: 'input',
-      name: 'iAppVersion',
-      message: 'What is the version of your iApp?',
-      default: '0.1.0',
-    },
-  ]);
+    const walletAddress = await askForWalletAddress({ spinner });
 
-  const walletAddress = await askForWalletAddress({ spinner });
+    // if an app secret must be set we will need the app owner wallet to push it
+    let iexec;
+    if (appSecret !== null) {
+      const privateKey = await askForWalletPrivateKey({ spinner });
```

Comment:
Asking for the private key can be a blocker. At least we need to explain why we need it?
Add a spinner info: "To push an app secret, you need to sign a transaction, hence asking for your private key to do it on your behalf."
(Not sure about this exact sentence)

Reply:
I'm not sure I want to address the problem in this PR, but an option could be to generate a wallet for the users when running `app init` and never ask them to import their wallet. An option for importing wallets for advanced users can be added as well.

Reply:
fa38dc0b4a1c694dbd7d6c77596b0d9b179a6058